tests/serialization/test_dag_serialization.py

```diff
@@ -67,14 +69,49 @@
                 "ui_color": "#fff",
                 "ui_fgcolor": "#000",
                 "template_fields": [],
+                "extra_links": [],
                 "_task_type": "BaseOperator",
             },
+            "custom_task": {
+                "task_id": "custom_task",
+                "retries": 1,
+                "retry_delay": 300.0,
+                "_downstream_task_ids": [],
+                "_inlets": {
+                    "auto": False, "task_ids": [], "datasets": []
+                },
+                "_outlets": {"datasets": []},
+                "ui_color": "#fff",
+                "ui_fgcolor": "#000",
+                "template_fields": [],
+                "extra_links": ["github"],
```

Comment:
I'm not sure the Link name is unique enough -- Given the `link.name` attribute is what is displayed in the UI it is entirely reasonable for two different link classes to define a link with a name of "Logs" or "View in AWS" but that have different implementations.

I think `<plugin_name>.<LinkClass>` would be better, so `test_plugin.GithubLink`.

(To make this work we'd likely need to change a bit how we store plugins in airflow.plugins_manager)

Reply:
I have updated it with a different approach: https://github.com/coufon/airflow/compare/d7bc655..39b5b7c?expand=1

Also created a separate PR for it https://github.com/apache/airflow/pull/6302